### PR TITLE
Remove jvm log4cats dependency in ember client

### PR DESCRIPTION
**Pull request by kubukoz**
Backport of #5757.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -24,3 +24,5 @@ addSbtPlugin("org.planet42" % "laika-sbt" % "0.18.1")
 // TODO remove me after we get this transitively
 // https://github.com/djspiewak/sbt-github-actions/issues/94
 addSbtPlugin("com.codecommit" % "sbt-github-actions" % "0.14.2")
+
+addDependencyTreePlugin
```
Comment (kubukoz) [inline project/plugins.sbt:28]: I can remove this if needed, but it was helpful here and might be helpful to you in the future.

File: build.sbt [hunk 2]
```diff
@@ -391,7 +391,7 @@ lazy val client = libraryCrossProject("client")
       nettyCodecHttp % Test
     )
   )
-  .dependsOn(core, server, testing % "test->test", theDsl % "test->compile")
+  .dependsOn(core, server % "test", testing % "test->test", theDsl % "test->compile")
```
Comment (armanbilge) [inline build.sbt:394]: This one is 💯 on me in https://github.com/http4s/http4s/pull/5563, sorry everyone 😳

Comment (armanbilge) [reply build.sbt:394]: while I'm here 😉 

```suggestion
  .dependsOn(core, server % Test, testing % "test->test", theDsl % "test->compile")
```

**Comment by armanbilge** [comment]
Retargeted in https://github.com/http4s/http4s/pull/5757.

**Pull request closed without merging**